deepchem/feat/molecule_featurizers/dmpnn_featurizer.py

```diff
@@ -254,3 +261,226 @@ def map_reac_to_prod(
+  - Aromatic: A one-hot vector of whether the atom belongs to an aromatic ring.
+  - Mass: Atomic mass * 0.01
+
+  The default edge representation are constructed by concatenating the following values,
+  and the feature length is 14.
+
+  - Bond type: A one-hot vector of the bond type, "single", "double", "triple", or "aromatic".
+  - Same ring: A one-hot vector of whether the atoms in the pair are in the same ring.
+  - Conjugated: A one-hot vector of whether this bond is conjugated or not.
+  - Stereo: A one-hot vector of the stereo configuration (0-5) of a bond.
+
+  If you want to know more details about features, please check the paper [1]_ and
+  utilities in deepchem.utils.molecule_feature_utils.py.
+
+  Examples
+  --------
+  >>> smiles = ["C1=CC=CN=C1", "C1CCC1"]
+  >>> featurizer = DMPNNFeaturizer()
+  >>> out = featurizer.featurize(smiles)
+  >>> type(out[0])
+  <class 'deepchem.feat.graph_data.GraphData'>
+  >>> out[0].num_node_features
+  133
+  >>> out[0].node_features.shape
+  (6, 133)
+  >>> out[0].node_features_zero_padded.shape
+  (7, 133)
+  >>> out[0].num_edges
+  12
+  >>> out[0].concatenated_features_zero_padded.shape
+  (13, 147)
+  >>> len(out[0].mapping)
+  13
+
+  References
+  ----------
+  .. [1] Kearnes, Steven, et al. "Molecular graph convolutions: moving beyond fingerprints."
+     Journal of computer-aided molecular design 30.8 (2016):595-608.
+
+  Note
+  ----
+  This class requires RDKit to be installed.
+  """
+
+  def __init__(self,
+               features_generator=None,
+               is_adding_hs: bool = False,
+               use_original_atom_ranks: bool = False,
+               features_scaling: bool = False,
+               atom_descriptor_scaling: bool = False,
+               bond_feature_scaling: bool = False) -> None:
+    """
+    Parameters
+    ----------
+    features_generator:
+      # TODO: Implement global feature generator
+    is_adding_hs: bool, default False
+      Whether to add Hs or not.
+    use_original_atom_ranks: bool, default False
+      Whether to use original atom mapping or canonical atom mapping
+    features_scaling: bool, default False
+      # TODO: feature normalization
+    atom_descriptor_scaling: bool, default False
+      # TODO: descriptor normalization
+    bond_feature_scaling: bool, default False
+      # TODO: bond feature normalization
+    """
+    self.features_generator = features_generator
+    self.is_adding_hs = is_adding_hs
+    self.use_original_atom_ranks = use_original_atom_ranks
+    self.features_scaling = features_scaling
+    self.atom_descriptor_scaling = atom_descriptor_scaling
+    self.bond_feature_scaling = bond_feature_scaling
+    self.atom_descriptors = None
+    self.phase_features = None
+
+  def _featurize(self, datapoint: RDKitMol, **kwargs) -> GraphData:
+    """Calculate molecule graph features from RDKit mol object.
+
+    Parameters
+    ----------
+    datapoint: rdkit.Chem.rdchem.Mol
```

Comment:
```python3
RDKitMol
``` since that's what you're using in the argument declaration.

Reply:
ok noted. I will make the required changes